### PR TITLE
fix(cache): use UTC dates for cache strategy calculations

### PR DESCRIPTION
**Pull request by cgrindel**
## Changes

Use UTC dates and deterministic durations for cache strategy date calculations.

## Context

The fix #21389 appears to have broken unit tests for developers in [CEST](https://en.wikipedia.org/wiki/Central_European_Summer_Time) and probably other timezones with a positive offset. 

I believe that the underlying issue is that the current implementation performs date calculations in the current system timezone. This can lead to ambiguities when calculating expiration timestamps.  

In addition, I found another date calculation that was using `days` for its date calculation. I have switched it to `hours` like the fix in #21389.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — DISMISSED** [review]

**Comment by cgrindel** [comment]
Please don't squash yet. I am still running tests locally with different timezones.

**Comment by viceice** [comment]
we won't do as long as the PR is in draft. 😏

**Comment by cgrindel** [comment]
Tested locally by running the unit tests in MST and CEST. Both runs are green.

**Review by viceice — APPROVED** [review]

File: lib/util/github/graphql/cache-strategies/abstract-cache-strategy.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export abstract class AbstractGithubGraphqlCacheStrategy<
   /**
    * The time which is used during single cache access cycle.
    */
-  protected readonly now = DateTime.now();
+  protected readonly now = DateTime.now().toUTC();
```
Comment (NiasSt90) [inline lib/util/github/graphql/cache-strategies/abstract-cache-strategy.ts:26]: `DateTime.utc()`?

Comment (cgrindel) [reply lib/util/github/graphql/cache-strategies/abstract-cache-strategy.ts:26]: Looking at [the Luxon doc](https://moment.github.io/luxon/api-docs/index.html), I see `toUTC()`? I do not see an instance method called `utc()`. I do see a static function called `utc()`.

**Pull request merged**